go.mod

```diff
@@ -1,6 +1,7 @@
 module github.com/containerd/containerd/v2
 
 go 1.22.0
+toolchain go1.23.4
```

Comment:
Since this doesnt change the minimum version when using the module, this change should be fine. 

But should make sure that it doesnt affect projects which import containerd. cc: @thaJeztah to see if it is fine with moby.